### PR TITLE
[HOPSWORKS-954] Add support for ROCm

### PR DESCRIPTION
**Pull request by robzor92**

### CODE REVIEW

**Comment by robzor92** [comment]
https://github.com/logicalclocks/hops-testing/pull/360

**Comment by robzor92** [comment]
@tkakantousis @SirOibaf please review

**Comment by robzor92** [comment]
@kouzant please review.

File: files/default/agent.py [hunk 1]
```diff
@@ -236,7 +224,7 @@ def send(self):
                 now = long(time.mktime(datetime.now().timetuple()))
                 headers = {'content-type': 'application/json'}
                 payload = {}
-                payload["num-gpus"] = count_num_gpus()
+                payload["num-gpus"] = devices.get_num_gpus()
```
Comment (kouzant) [inline files/default/agent.py]: It is important that 'get_num_gpus' does not raise any exception that could kill the heartbeat. Otherwise I'm OK with PR

Comment (robzor92) [reply files/default/agent.py]: This should be reliable enough: https://github.com/logicalclocks/hops-util-py/blob/bbb4232036ee8057d57b4d472a49b715cc22771c/hops/devices.py#L115

**Pull request merged**